### DISCUSSION TITLE
Retain partition configuration across restarts ?

### DISCUSSION THREAD

**Discussion by jcorporation** [MusicPlayerDaemon/MPD#1520] [Q&A]
Playing with the latest mpd version with the partition feature. It works all as documented, but partition configuration is lost across restarts.

Is this a bug or a not implemented feature?

**Comment by tsunulukai** [comment]
I experienced the same issue while trying mpd paritition feature.
Loosing partition state across mpd restarts is quite annoying...

**Comment by sevmonster** [comment]
I do not think it is a bug, because the feature appears not implemented. From what I can see, partitions are [supported by `StateFile`s](https://github.com/MusicPlayerDaemon/MPD/blob/89a18b49a7c2929591e1dc98719b12cd8874d180/src/StateFile.hxx#L40), but MPD only uses [the current partition for the instance's state file at startup](https://github.com/MusicPlayerDaemon/MPD/blob/89a18b49a7c2929591e1dc98719b12cd8874d180/src/Main.cxx#L268), which is always the default partition. Because of this, only the default partition ever has its state saved, and MPD will be ignorant of any other partitions' state.

While there is no facility to read or write state files for other partitions, all the code needed to support it appears to be there. Maybe a list of extra state files to read for each partition could be included in the main state file (i.e. `mpd/state` has `partitions: hello,world` and `mpd/partitions/` directory contains state files `hello` and `world`), or shunt them all into a folder named after their partition (i.e. `mpd/state` → `mpd/states/default`, partitions `hello` and `world` write to state files `mpd/states/{hello,world}`)

For the last suggestion, the default state file `mpd/state` could be retained backwards compatibility, possibly without the `partition` line.

**Comment by john-** [comment]
I am running mpd version 0.23.11 and also see partitions not being saved across mpd restarts (using mpMPD as a client).  In addition, when I manually add this to mpd.conf then mpd does not start:
```
partition {                                                                                        
    name "mobile"                                                                                  
}                                                                                                  
```

I am not sure if that is related or expected behavior.

**Reply by jcorporation** [reply]
This config options is only supported since MPD 0.24

**Reply by john-** [reply]
Good to know....thanks for the information.
